apps/src/blockTooltips/DropletFunctionTooltip.js

```diff
@@ -0,0 +1,59 @@
+var DropletBlockTooltipMarkup = require('./DropletBlockTooltip.html');
+var msg = require('../../locale/current/common');
+
+/**
+ * @fileoverview Representation of a droplet function/block's tooltip
+ */
+'use strict';
+
+/**
+ * Stores a block's tooltip information and helps render it
+ * @constructor
+ */
+var DropletFunctionTooltip = function (functionName) {
+  /** @type {String} */
+  this.functionName = functionName;
+
+  /** @type {String} */
+  this.description = null;
+
+  if (msg.hasOwnProperty(this.descriptionKey())) {
+    this.description = msg[this.descriptionKey()]();
+  }
+
+  /** @type {Array.<String>} */
+  this.paramNames = [];
+
+  var paramId = 0;
+  while (msg.hasOwnProperty(this.parameterKey(paramId))) {
```

Comment:
nit:
var paramKey = this.parameterKey(paramId);
Then you only need to make the call once.